### DISCUSSION TITLE
[error] Could not create schema migrations table

### DISCUSSION THREAD

**Discussion by mgh520** [TomGrozev/arangox_ecto#51] [General]
Hi all. I'm very new to arangodb and this ecto project.
I'm working on a simple phoenix live-view sample project and using the arangox_ecto adapter.
I am using the default "dynamic" mode so my understanding is there shouldn't be a migration table or migrations.

The code reloading feature of phoenix will trigger this error about the migrations table: "[error] Could not create schema migrations table" on every reload.

The problem comes from the last line of this code generated in the project's `endpoint.ex` file:

```
if code_reloading? do
    socket "/phoenix/live_reload/socket", Phoenix.LiveReloader.Socket
    plug Phoenix.LiveReloader
    plug Phoenix.CodeReloader
    plug Phoenix.Ecto.CheckRepoStatus, otp_app: :phx_arango
  end

```

Here are the versions of things in my project:

      {:arangox_ecto, "~> 1.2"},
      {:phoenix, "~> 1.6.13"},
      {:phoenix_ecto, "~> 4.4"},
      {:ecto_sql, "~> 3.6"},
      {:postgrex, ">= 0.0.0"},
      {:phoenix_html, "~> 3.0"},
      {:phoenix_live_reload, "~> 1.2", only: :dev},
      {:phoenix_live_view, "~> 0.17.5"},

I can simply remove the `plug Phoenix.Ecto.CheckRepoStatus, otp_app: :phx_arango` line and it won't check anymore, but is there something different I should be doing to avoid this error?  Is there a way the arangox_ecto project should tie into this behavior to avoid it?  It doesn't affect functionality, it just spits a giant annoying error in the terminal whenever a file is changed.

thanks in advance!

**Comment by TomGrozev** [comment]
Have you run `mix ecto.setup.arango` first? The reason that that error would show is if your database doesn't exist.
